autolens/analysis/aggregator/aggregator.py

```diff
@@ -229,6 +174,261 @@ def fit_interferometer_via_database_from(
     )
 
 
+class AbstractAgg:
+    def __init__(self, aggregator: af.Aggregator):
+        """
+        An abstract aggregator wrapper, which makes it straight forward to compute generators of objects from specific
+        samples of a non-linear search.
+
+        For example, in **PyAutoLens**, this makes it straight forward to create generators of `Tracer`'s drawn from
+        the PDF estimated by a non-linear for efficient error calculation of derived quantities.
+        Parameters
+        ----------
+        aggregator
+            An PyAutoFit aggregator containing the results of non-linear searches performed by PyAutoFit.
+        """
+        self.aggregator = aggregator
+
+    def make_object_for_gen(self, fit: Fit, galaxies: List[al.Galaxy]) -> object:
+        """
+        Abstract method that is overwritten with functions that create the generator of an object from the results
+        of a non-linear search.
+
+        For example, in the `TracerAgg` object, this function is overwritten such that it creates a `Tracer` from a
+        `ModelInstance` that contains the galaxies of a sample from a non-linear search.
+
+        Parameters
+        ----------
+        fit
+            A PyAutoFit database Fit object containing the generators of the results of PyAutoGalaxy model-fits.
+        galaxies
+            A list of galaxies corresponding to a sample of a non-linear search and model-fit.
+
+        Returns
+        -------
+        Generator
+            A generator that creates an object used in the model-fitting process of a non-linear search.
+        """
+        raise NotImplementedError
+
+    def max_log_likelihood_gen(self) -> Generator:

```

Comment:
You can actually specify the type of object returned by the generator although it's a little ugly because it accounts for coroutines.

Something like:
```python
Generator[float, None, None]